preview/Win7Msix/Win7MSIXInstaller/ProcessPotentialUpdate.cpp

```diff
@@ -0,0 +1,57 @@
+#include "ProcessPotentialUpdate.hpp"
+#include <filesystem>
+
+const PCWSTR ProcessPotentialUpdate::HandlerName = L"ProcessPotentialUpdate";
+
+HRESULT ProcessPotentialUpdate::ExecuteForAddRequest()
+{
+    std::wstring currentPackageFamilyName = GetFamilyNameFromFullName(m_msixRequest->GetPackageInfo()->GetPackageFullName());
+    
+    for (auto& p : std::experimental::filesystem::directory_iterator(m_msixRequest->GetFilePathMappings()->GetMsix7Directory()))
+    {
+        std::wstring installedPackageFamilyName = GetFamilyNameFromFullName(p.path().filename());
+        if (CaseInsensitiveEquals(currentPackageFamilyName, installedPackageFamilyName)
+            && !CaseInsensitiveEquals(m_msixRequest->GetPackageInfo()->GetPackageFullName(), p.path().filename()))
+        {
+            TraceLoggingWrite(g_MsixTraceLoggingProvider,
+                "Found an update to an existing package, removing package",
+                TraceLoggingValue(p.path().filename().c_str(), "PackageToBeRemoved"));
+
+            RETURN_IF_FAILED(RemovePackage(p.path().filename()));
+            return S_OK;
+        }
+    }
+
+    TraceLoggingWrite(g_MsixTraceLoggingProvider,
+        "Not an update, nothing to do.");
+    return S_OK;
+}
+
+HRESULT ProcessPotentialUpdate::RemovePackage(std::wstring packageFullName)
+{
+    AutoPtr<MsixRequest> localRequest;
+    RETURN_IF_FAILED(MsixRequest::Make(OperationType::Remove, Flags::NoFlags, std::wstring(), packageFullName, MSIX_VALIDATION_OPTION::MSIX_VALIDATION_OPTION_FULL, &localRequest));

```

Comment:
>MsixRequest [](start = 21, length = 11)

I would recommend adding a log here, as this is a remove that we are creating as part of update. So it will be easier to debug through these cases.

Reply:
moved the log from ExecuteForAddRequest to here

---
In reply to: [269347604](https://github.com/Microsoft/msix-packaging/pull/58#discussion_r269347604) [](ancestors = 269347604)